### DISCUSSION TITLE
Klarna integration with Saleor

### DISCUSSION THREAD

**Discussion by satvikt** [saleor/saleor#8962] [General]
We are developing an e-commerce website using Saleor and wish to integrate **Klarna Checkout** and **Klarna Shipping Assistant** with it.

A brief of the above:

**Klarna Checkout** ([link](https://docs.klarna.com/klarna-checkout/)) is a payment solution where Klarna handles your store’s entire checkout. This solution includes all of Klarna’s payment methods: Pay now (card payments), Pay in 30 days (invoice), Pay in 3 instalments and Financing (instalment plans of multiple months).

**Klarna Shipping Assistant** ([link](https://docs.klarna.com/shipping-assistant/)) is the shipping selector for the **Klarna Checkout**.

I have looked at Saleor integrations with Payment Gateways like Adyen, RazorPay etc. However, integrating Klarna Checkout involves providing a whole new Checkout experience.
 
**Has someone worked on a similar use case/has tips for the same?**

Other useful links:
1. Klarna Checkout API - https://docs.klarna.com/klarna-checkout/api/checkout-api/
2. Klarna Shipping API integration - https://docs.klarna.com/shipping-assistant/api-integration/

**Comment by korycins** [comment]
Hey @satvikt 
I don't think that we have any experience in integrating Klarna directly. You can use other payment gateways as an example, like: StripePlugin or AdyenPlugin. 
Also, not sure if you already know that, but we support some Klarna's payment methods via AdyenPlugin (Adyen offers Klarna's payment methods as one of option).

**Reply by satvikt** [reply]
Hi @korycins,
I tried to use the Adyen Plugin by enabling it in the dashboard and filling the required values. I observed that while the `react-storefront` fetches it as one of the payment gateways (saw this in the graphql response in Chrome network tab), it is unable to render the "Adyen payment gateway" on the actual site. 

**What's the mechanism to enable the Adyen payment gateway on the storefront?**

_(PS: On careful inspection, I saw that while "Dummy" and "Stripe" payment gateways had their respective code blocks in the `react-storefront` repo, there wasn't any corresponding to "Adyen". Do we have to write custom code for it? Or is something for Adyen in the pipeline?)_

**Reply by korycins** [reply]
Hey @satvikt,

Here you can find a PR for missing Adyen's docs - https://github.com/saleor/saleor-docs/pull/340 It is still WIP but could be helpful.
AFAIK react-storefront doesn't support an Adyen as a payment gateway yet. You can check the implementation dedicated for Adyen in the old one - https://github.com/saleor/saleor-storefront.
